jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c

```diff
@@ -2664,6 +2666,206 @@ ecma_builtin_array_prototype_includes (const ecma_value_t args[], /**< arguments
+{
+  uint32_t target_index = start;
+
+  for (uint32_t source_index = 0; source_index < source_len; source_index++)
+  {
+    ecma_value_t element = ecma_op_object_find_by_uint32_index (source, source_index);
+
+    if (ECMA_IS_VALUE_ERROR (element))
+    {
+      return element;
+    }
+
+    if (!ecma_is_value_found (element))
+    {
+      continue;
+    }
+
+    if (!ecma_is_value_undefined (mapped_value))
+    {
+      ecma_value_t source_val = ecma_make_uint32_value (source_index);
+      ecma_value_t args[] = {element,  source_val, ecma_make_object_value (source)};
+      ecma_value_t temp_element = ecma_op_function_call (ecma_get_object_from_value (mapped_value), thisArg, args, 3);
+
+      ecma_free_value (element);
+      ecma_free_value (source_val);
+
+      if (ECMA_IS_VALUE_ERROR (temp_element))
+      {
+        return temp_element;
+      }
+
+      element = temp_element;
+    }
+
+    if (depth > 0)
+    {
+      ecma_value_t is_array = ecma_is_value_array (element);
+
+      if (ECMA_IS_VALUE_ERROR (is_array))
+      {
+        ecma_free_value (element);
+        return is_array;
+      }
+
+      if (ecma_is_value_true (is_array))
+      {
+        ecma_object_t *element_obj = ecma_get_object_from_value (element);
+        uint32_t element_len;
+        ecma_value_t len_value = ecma_op_object_get_length (element_obj, &element_len);
+
+        if (ECMA_IS_VALUE_ERROR (len_value))
+        {
+          ecma_deref_object (element_obj);
+          return len_value;
+        }
+
+        ecma_value_t target_index_val = ecma_builtin_array_flatten_into_array (target,
+                                                                               element_obj,
+                                                                               element_len,
+                                                                               target_index,
+                                                                               depth - 1,
+                                                                               ECMA_VALUE_UNDEFINED,
+                                                                               ECMA_VALUE_UNDEFINED);
+
+        ecma_deref_object (element_obj);
+
+        if (ECMA_IS_VALUE_ERROR (target_index_val))
+        {
+          return target_index_val;
+        }
+
+        target_index = (uint32_t) ecma_get_number_from_value (target_index_val);
+        JERRY_ASSERT (target_index != UINT32_MAX);
+        continue;
+      }
+    }
+
+    const uint8_t flags = ECMA_PROPERTY_CONFIGURABLE_ENUMERABLE_WRITABLE;
+    ecma_value_t element_temp = ecma_builtin_helper_def_prop_by_index (ecma_get_object_from_value (target),
+                                                                       target_index,
+                                                                       element,
+                                                                       flags);
+
```

Comment:
FlattenIntoArray could throw, flags should be ECMA_PROPERTY_CONFIGURABLE_ENUMERABLE_WRITABLE | ECMA_IS_THROW. / https://www.ecma-international.org/ecma-262/11.0/#sec-flattenintoarray /

Reply:
True, thank you for pointing it out.